src/components/Charts/mutationsTree.js

```diff
@@ -0,0 +1,65 @@
+
+import React from 'react';
+import Container, {Title} from "./styles";
+
+class MutationsTree extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {};
+    }
+
+    createVariantString(variants, level){
+        let newText="";
+        let first=true;
+        for(const variant of variants){
+            console.log(variant);
+            console.log(variant.name);
+            newText+='---'.repeat(level)+"variant: "+ variant.name+" (";
+            first=true;
+            for(const mutation of variant.mutations){
+                if(!first){
+                    newText+=", "+mutation.from+mutation.position+mutation.to;
+                }
+                else{
+                    newText+=mutation.from+mutation.position+mutation.to;
+                    first=false
+                }
+            }
+            newText+=")\n";
+            if(variant.subs.length>0){
+                newText+=this.createVariantString(variant.subs, level+1);
+            }
+        }
+        return newText;
+    }
+
+
+    componentDidMount() {
+        console.log("props data", this.props.data);
+        let newText=this.createVariantString(this.props.data, 0);
+        let newLinesToPtagText = newText.split('\n').map(i => {
```

Comment:
why not just return array of variants in the first place?

Reply:
How exactly? There was a problem with new lines - we did not see the line breaks, hence we added lines 39-41 to wrap lines in <p> tags. Then we use the formatted text in render.

Reply:
change createVariantString => createVariants and the function would produce string[], one string per line.
 As far as I can understand, it creates newlines internally and it could just append to array instead. (Though, I am not sure whether are aren't any other newlines in the interpolated strings.